tests/integration/test_basket_waitlist_subscription.py

```diff
@@ -0,0 +1,236 @@
+    (
+        f"{settings.basket_server_url}/readiness/",
+        f"{settings.basket_server_url}/healthz/",
+        f"{settings.ctms_server_url}/__heartbeat__",
+    ),
+)
+def test_connectivity(url):
+    resp = requests.get(url)
+    resp.raise_for_status()
+
+
+def test_vpn_waitlist(ctms_headers):
+    # 1. Subscribe a certain email to the `vpn` waitlist
+    email = f"integration-test-{uuid4()}@restmail.net"
+    waitlist = "guardian-vpn-waitlist"
+
+    basket_subscribe(email, waitlist, fpn_country="us", fpn_platform="ios,android")
+
+    # 2. Basket should have set the `vpn_waitlist` field/data.
+    # Wait for the worker to have processed the request.
+    time.sleep(2)
+    contact_details = ctms_fetch(email, ctms_headers)
+    assert contact_details["waitlists"] == [
+        {
+            "name": "vpn",
+            "source": None,
+            "fields": {
+                "geo": "us",
+                "platform": "ios,android",
+            },
+        }
+    ]
+    assert contact_details["vpn_waitlist"] == {
+        "geo": "us",
+        "platform": "ios,android",
+    }
+
+    # 3. Patch an attribute (eg. change country)
+    email_id = contact_details["email"]["email_id"]
+    resp = requests.patch(
+        f"{settings.ctms_server_url}/ctms/{email_id}",
+        headers=ctms_headers,
+        json={
+            "vpn_waitlist": {"geo": "fr", "platform": "linux"},
+        },
+    )
+    resp.raise_for_status()
+    # Request the full contact details again.
+    contact_details = ctms_fetch(email, ctms_headers)
+    assert contact_details["vpn_waitlist"] == {
+        "geo": "fr",
+        "platform": "linux",
+    }
+
+    newsletters_by_name = {nl["name"]: nl for nl in contact_details["newsletters"]}
+    assert newsletters_by_name[waitlist]["subscribed"]
+
+    # 4. Unsubscribe from Basket
+    basket_token = contact_details["email"]["basket_token"]
+    basket_unsubscribe(basket_token, waitlist)
+
+    # Request the full contact details again.
+    # Wait for the worker to have processed the request.
+    time.sleep(2)
+    contact_details = ctms_fetch(email, ctms_headers)
+
+    newsletters_by_name = {nl["name"]: nl for nl in contact_details["newsletters"]}
+    assert not newsletters_by_name[waitlist]["subscribed"]
+    assert contact_details["vpn_waitlist"] == {
+        "geo": None,
+        "platform": None,
+    }
+
+
+def test_relay_waitlists(ctms_headers):
+    email = f"stage-test-{uuid4()}@restmail.net"
+    waitlist = "relay-waitlist"
+
+    # 1. Subscribe a certain email to the `relay` waitlist
+    basket_subscribe(
+        email, waitlist, relay_country="es", source_url="https://relay.firefox.com/"
+    )
+
+    # 2. Basket should have set the `relay_waitlist` field/data.
+    # Wait for the worker to have processed the request.
+    time.sleep(2)
+    contact_details = ctms_fetch(email, ctms_headers)
+
+    # 3. CTMS should show both formats (legacy `relay_waitlist` field, and entry in `waitlists` list)
```

Comment:
So does this mean basket will be sending both formats for a period of time? 
Or is CTMS manipulating the data into two forms upon request?

Reply:
[I think I understand](https://github.com/mozmeao/basket/pull/962/files#diff-d6aadfae5982cb77055ddff406cd982f1dce929c139d9dd562ca488f0b9ad568R216), please correct me if this is mistaken:
- Basket sends both forms if the waitlist is from a pre-existing waitlist;
- otherwise Basket only sends the new format to CTMS

Reply:
I'm seeing these tests: `test_vpn_waitlist` and `test_relay_waitlists`
They are asserting on _both_ the legacy and new format.
Is there any desire/need/possibility for an additional test that confirms that the legacy format will _not be sent_ with "new" waitlists?

Reply:
> So does this mean basket will be sending both formats for a period of time?

Current Basket reads and sends `*-waitlist` newsletters + vpn/relay waitlists as contact attributes.
CTMS (`main` branch) stores both, newsletters AND waitlists records.

Future Basket reads and sends from waitlists only.
`*-waitlist` newsletters can be deleted from CTMS.

Reply:
> 
>   Is there any desire/need/possibility for an additional test that confirms that the legacy format will not be sent with "new" waitlists?
>

This integration test leverages the old basket still

Once Basket is released and we can update the container of this test, yes, we can adjust it https://github.com/mozilla-it/ctms-api/issues/605